internal/httpcore/httpcore.go

```diff
@@ -55,6 +57,7 @@ func ListenAndServe(ctx context.Context, addr string) error {
 	})
 
 	e := endpoints{}
+	m := middleware{}
 
 	app.Use(func(ctx *fiber.Ctx) error {
 		authToken := ctx.Cookies(sessionTokenCookieName)
```

Comment:
Since the only purpose of this middleware is to set the `isauthset` and `token` locals, which are then only used by `httpcore.getSessionAuth`, which in turn is only used in `httpcore.middleware.requireAuthenticated`, it might be worth rolling all three of these things (or maybe keeping `getSessionAuth` separate for readability) into `requireAuthenticated` to reduce sprawliness.

Reply:
Done this, but also moved the functionality of `validateUserExists` into `requireAuthenticated`, since `getSessionAuth` (now `getAuthStatus`) already reads the user table and checks if there are any results